### ISSUE TITLE
QAT using pytorch-quantization cause accuracy loss after exporting to onnx

### ISSUE THREAD

**Issue by MomoP32** [NVIDIA/TensorRT#3654]
## Description

  I use pytorch-quantization to do QAT for a pointpillar model,  it works fine during pytorch training, however, when I export the torch model to onnx, accuracy degrades badly. In my opinion, this process should have no loss of accuracy at all.

## Environment

**TensorRT Version**:
8.5.2
**NVIDIA GPU**:
RTX3060 
**NVIDIA Driver Version**:
470.161.03
**CUDA Version**:
11.3
**CUDNN Version**:
8.6

Operating System:
Ubuntu 20.04
Python Version (if applicable):
3.8
PyTorch Version (if applicable):
1..11
pytorch-quantization:
2.1.2

## Relevant Files
here is my codes to export onnx
![T7TIq2wocc](https://github.com/NVIDIA/TensorRT/assets/127090509/7607d4a1-bcf1-4a46-849d-bf82228d9741)

**Comment by zerollzeng** [comment]
Yes, should be a bug of pytorch, could you please file bug against pytorch repo?

**Comment by zerollzeng** [comment]
@ttyio for viz.

**Comment by MomoP32** [comment]
> Yes, should be a bug of pytorch, could you please file bug against pytorch repo?

sure, I'll do it; actually, [I gave up the method of using explicit quantization, but introduced fake-quantization operators during training, and use the PTQ method to quant, and the result is okay.

**Comment by zerollzeng** [comment]
Closed.

**Comment by lucy3589** [comment]
> > Yes, should be a bug of pytorch, could you please file bug against pytorch repo?
> 
> sure, I'll do it; actually, [I gave up the method of using explicit quantization, but introduced fake-quantization operators during training, and use the PTQ method to quant, and the result is okay.

could you tell me how to introduced fake-quantization? 
Like this? Adding these before network initialization?

from pytorch_quantization import nn as quant_nn
from pytorch_quantization import quant_modules
quant_nn.TensorQuantizer.use_fb_fake_quant = True
quant_modules.initialize()

**Comment by MomoP32** [comment]
I have solved this problem;  During PTQ using TRT,  static quantization is executed, which means scale values will be determined. 
However, using pytorch_quantization to QAT our model, default setting leads dynamic quantization, which means scale values will change with input tensor.  So, when exporting onnx,  different  input values lead different scale for D/DQ of inputs.(conv weghts are fine).   If your model is based on image with max value 255,  this way is fine, but for lidar based,  this way failed.    Related codes,  default_input_quant_descriptor = QuantDescriptor(num_bits=8, name="cool quant", learn_amax=True), replace default input quant descriptor with this one, openning learn_amax.    All inputs scale values will be determined by training, instead of one input.  
Also, I recommend add one sample in official document to clarify dynamic and static quantization.
